core/src/main/kotlin/fr/sncf/osrd/api/pathfinding/constraints/SignalingSystemConstraints.kt

```diff
@@ -0,0 +1,47 @@
+package fr.sncf.osrd.api.pathfinding.constraints
+
+import fr.sncf.osrd.graph.EdgeToRangesId
+import fr.sncf.osrd.graph.Pathfinding
+import fr.sncf.osrd.signaling.SignalingSimulator
+import fr.sncf.osrd.sim_infra.api.*
+import fr.sncf.osrd.train.RollingStock
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.meters
+
+@JvmRecord
+data class SignalingSystemConstraints(
+    val blockInfra: BlockInfra,
+    val signalingSimulator: SignalingSimulator,
+    val rollingStocks: Collection<RollingStock>
+) : EdgeToRangesId<Block> {
+    override fun apply(edge: BlockId): MutableCollection<Pathfinding.Range<Block>> {
+        val res = HashSet<Pathfinding.Range<Block>>()
+        for (rs in rollingStocks) {
+            res.addAll(getBlockedRanges(rs, edge, blockInfra, signalingSimulator))
+        }
+        return res
+    }
+
+    companion object {
+        /**
+         * Returns the sections of the given block that can't be used by the given rolling stock
+         */
+        private fun getBlockedRanges(
+            stock: RollingStock,
+            edge: BlockId,
+            blockInfra: BlockInfra,
+            signalingSimulator: SignalingSimulator
+        ): Set<Pathfinding.Range<Block>> {
+            val blockSigSystem = blockInfra.getBlockSignalingSystem(edge)
+            val rsSupportedSigSystems = stock.SupportedSignalingSystems.map { s -> signalingSimulator.sigModuleManager.findSignalingSystem(s) }
```

Comment:
This is something that should be done when initializing the constraint (you can make a constructor or builder).

This will throw a `NullPointerException` if a signaling system isn't recognized, I don't think it will be a problem soon but it's not the best practice. We could throw a more explicit error, or possibly just ignore the unrecognized ones (that would allow us to have more data than just what core can handle)

Reply:
added a factory for this constraint and a catch if signaling system is not found

```
class SignalingSystemConstraintsFactory {
    fun create(
        blockInfra: BlockInfra,
        signalingSimulator: SignalingSimulator,
        rollingStocks: Collection<RollingStock>,
    ): SignalingSystemConstraints {
        val rsSupportedSigSystems = rollingStocks.map { stock ->
            stock.supportedSignalingSystems.mapNotNull { s -> try {
                signalingSimulator.sigModuleManager.findSignalingSystem(s)
            } catch (e: Exception) {
                null
            } }
        }
        return SignalingSystemConstraints(
            blockInfra,
            rsSupportedSigSystems
        )
    }
}
```